File: 164-markdown-inline-code-parser/solve.py
def parse_inline_code(markdown):
    is_open = False
    result = []

    for char in markdown:
        if char == "`":
            is_open = not is_open
            result.append("<code>" if is_open else "</code>") 
        else:
            result.append(char) 

    return ''.join(result)
    

File: 164-markdown-inline-code-parser/solve.js
/*
function parseInlineCode(markdown) {
  // let indexesOfTicks;
  let iteratorOddEven = 0;
  let newMarkdown = markdown;

  for (let i=0; i<=markdown.length-1; i++) {
    if (markdown[i] == "`") {
       // indexesOfTicks.push(i);
       iteratorOddEven+=1;
   
       if (iteratorOddEven%2==0) {
          newMarkdown.replace("`", "</code>")
       } else {
          newMarkdown.replace("`", "<code>")
       }
    }
  }
  return newMarkdown;
}
*/


function parseInlineCode(markdown) {
  let isOpen = false; 

  return markdown.split("").map(char => {
    if (char === "`") {
      isOpen = !isOpen; 
      return isOpen ? "<code>" : "</code>"; 
    }
    return char; 
  }).join(""); 
}
    